index.bs

```diff
@@ -195,11 +212,32 @@ When an <a>image request</a> has become <a>completely available</a> and decoded,
 
 <div algorithm="image element loaded">
     1. Let |image request| be the input <a>image request</a>.
-    1. If |image request|'s <a>associated element</a> is not <{img}>, <{image}>, or <a>background-image</a>, abort these steps.
-    1. If the <a>timing allow check</a> passes for |image request|'s resource, then add |image request| to <a>imagesPendingRendering</a>.
-    1. Otherwise, run the <a>report element timing</a> algorithm, passing in |image request| and 0.
+    1. Let |element| be |image request|'s <a>associated element</a>.
+    1. If |element| is not <{img}>, <{image}>, or <a>background-image</a>, return.
+    1. If the <a>timing allow check</a> fails for |image request|'s resource, then:
+        1. Run the <a>report image element timing</a> algorithm, passing in |image request| and 0.
```

Comment:
Reporting ET here for TAO failures and at "update the rendering" for TAO successes would result in weird ordering, no? Can we report all of them at the same place, but make sure "now" doesn't leak for TAO failures?

Reply:
Yes that is certainly an option, however this would leak |now| somewhat as follows:
Suppose you just have an image without TAO whose rendering timing you want to estimate. Only observe timing for that image. Then even though |now| is not exposed as a member of the entry you received, the fact that we wait until the update the rendering for the image means that we can distinguish between a very slow rendering time or a very fast rendering time by measuring the time at which we receive the entry.

The granularity of this attack is very bad since PerformanceObserver is not meant to dispatch its callback immediately, so maybe this is me being paranoid?

Reply:
Oh, that's clever! Yeah, misordering it is.
Can you add a note calling out that misordering specifically, and noting the reasons for it?

Reply:
Done, expanded on this on the Security section.